src/core/server/saved_objects/export/saved_objects_exporter.ts

```diff
@@ -118,28 +117,21 @@ export class SavedObjectsExporter {
     }: SavedObjectExportBaseOptions
   ) {
     this.#log.debug(`Processing [${savedObjects.length}] saved objects.`);
-    let exportedObjects: Array<SavedObject<unknown>>;
-    let missingReferences: SavedObjectsExportResultDetails['missingReferences'] = [];
 
-    savedObjects = await applyExportTransforms({
-      request,
+    const {
+      objects: collectedObjects,
+      missingRefs: missingReferences,
+    } = await collectExportedObjects({
       objects: savedObjects,
-      transforms: this.#exportTransforms,
-      sortFunction,
+      includeReferences: includeReferencesDeep,
+      namespace,
+      request,
+      exportTransforms: this.#exportTransforms,
+      savedObjectsClient: this.#savedObjectsClient,
     });
 
-    if (includeReferencesDeep) {
-      this.#log.debug(`Fetching saved objects references.`);
-      const fetchResult = await fetchNestedDependencies(
-        savedObjects,
-        this.#savedObjectsClient,
-        namespace
-      );
-      exportedObjects = sortObjects(fetchResult.objects);
-      missingReferences = fetchResult.missingRefs;
-    } else {
-      exportedObjects = sortObjects(savedObjects);
-    }
+    // sort with the provided sort function then with the default export sorting
+    const exportedObjects = sortObjects(collectedObjects.sort(sortFunction));
```

Comment:
This behavior is unchanged, the double sorting was just performed in two different steps